src/poetry_plugin_shell/command.py

```diff
@@ -40,7 +41,9 @@ def handle(self) -> int:
 
         # Be sure that we have the right type of environment.
         env = self.env
-        assert env.is_venv()
+        if not env.is_venv():
+            error_msg = "Virtual environment cannot be activated"
+            raise PoetryConsoleError(error_msg)
```

Comment:
Better off doing something like this instead:

```suggestion
        if not env.is_venv():
            self.line("")
            self.line_error("The Python environment at <info>{env.path}</> cannot be activated as it is not a virtural environment.")
            return 1
```

Reply:
applied, thanks for the suggestion, I didn't know about this method. it's much better